modules/examiner/php/NDB_Menu_Filter_Form_examiner.class.inc

```diff
@@ -0,0 +1,327 @@
+                               'radiologist' => 'COALESCE(e.radiologist, 0)',
+                              );
+
+        // If certification is turned on, update the class variables to include
+        // certification information
+        if ($useCertification == '1') {
+            array_push(
+                $this->columns,
+                'tn.Full_name as Certification',
+                'e.examinerID as ID',
+                'c.pass as Pass'
+            );
+            array_push($this->headers, 'Certification');
+            array_push($this->validFilters, 'tn.Full_name');
+            array_push($this->formToFilter, array('instrument' => 'tn.Full_name'));
+        }
+        return true;
+    }
+
+    /**
+     * Sets the menu filter class variables.
+     *
+     * @return void
+     */
+    function _setFilterForm()
+    {
+        $user = User::singleton();
+
+        // Get site options
+        if ($user->hasPermission('examiner_multisite')) {
+            // get the list of study sites - to be replaced by the Site object
+            $sites = Utility::getSiteList();
+            if (is_array($sites)) {
+                $sites = array('' => '') + $sites;
+            }
+        } else {
+            // allow only to view own site data
+            $site = Site::singleton($user->getData('CenterID'));
+            if ($site->isStudySite()) {
+                $sites = array($user->getData('CenterID') => $user->getData('Site'));
+            }
+        }
+
+        // Radiologist options
+        $radiologist = array(
+                        null => 'All',
+                        '1'  => 'Yes',
+                        '0'  => 'No',
+                       );
+
+        // menu filter form elements
+        $this->addBasicText('examiner', 'Examiner:');
+        $this->addSelect('site', 'Site:', $sites);
+        $this->addSelect('radiologist', 'Radiologist:', $radiologist);
+
+        // add examiner form elements
+        $this->addBasicText('addName', 'Name');
+        $this->addSelect('addSite', 'Site', $sites);
+
+        $this->form->addFormRule(array(&$this, '_validateAddExaminer'));
+
+        // Add a useCertification template variable so that the template
+        // rendering can depend on the EnableCertification setting
+        $config           = NDB_Config::singleton();
+        $certification    = $config->getSetting('Certification');
+        $useCertification = $certification['EnableCertification'];
+        $this->tpl_data['certification'] = $useCertification;
+
+    }
+
+    /**
+     * Sets the template data for the data table's rows
+     *
+     * @param int $count Page offset
+     *
+     * @return void
+     */
+    function _setDataTableRows($count)
+    {
+        // Code from old certification module
+        $config           = NDB_Config::singleton();
+        $certification    = $config->getSetting('Certification');
+        $useCertification = $certification['EnableCertification'];
```

Comment:
This seems to be done over and over again in the top of every function. Maybe it should be done once and assigned to a class variable in the setup function?